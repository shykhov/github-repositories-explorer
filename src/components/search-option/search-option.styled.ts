import styled from 'styled-components';

type OptionProps = {
  isFocused?: boolean;
};

export const OptionWrapper = styled.div<OptionProps>`
  cursor: pointer;
  padding: 7px 15px;
  display: flex;
  align-items: center;

  ${({ isFocused }) =>
    isFocused &&
    `
    background-color: #E6F2FF;
  `}

  &:hover {
    background-color: #e6f2ff;
  }

  & > span {
    margin-left: 10px;
    font-size: 16px;
  }
`;

export const ImageWrapper = styled.div`
  width: 25px;
  height: 25px;

  & > img {
    border-radius: 5px;
    max-width: 100%;
    max-height: 100%;
    object-fit: cover;
  }
`;
