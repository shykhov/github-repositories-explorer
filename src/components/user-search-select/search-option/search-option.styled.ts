import styled from 'styled-components';

type OptionProps = {
  isFocused?: boolean;
  isSelected?: boolean;
};

export const OptionWrapper = styled.div<OptionProps>`
  cursor: pointer;
  padding: 7px 0 7px 15px;
  display: flex;
  align-items: center;
  width: 100%;

  ${({ isSelected }) =>
    isSelected &&
    `
    + div {
      margin-left: 0;
    }
  `}

  ${({ isFocused }) =>
    isFocused &&
    `
    background-color: #E6F2FF;
  `}

  &:hover {
    background-color: #e6f2ff;
  }

  & > span {
    margin-left: 10px;
    font-size: 16px;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
    width: calc(100% - 40px);
  }
`;

export const ImageWrapper = styled.div`
  width: 25px;
  height: 25px;

  & > img {
    border-radius: 5px;
    max-width: 100%;
    max-height: 100%;
    object-fit: cover;
  }
`;
