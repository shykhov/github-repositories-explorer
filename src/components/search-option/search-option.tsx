import React from 'react';

import { OptionWrapper, ImageWrapper } from './search-option.styled';

export const SearchOption = (props: any) => {
  const { cx, innerProps, innerRef: ref, data, ...rest } = props;

  return (
    <OptionWrapper ref={ref} {...rest} {...innerProps}>
      {data && data.iconSrc && (
        <ImageWrapper>
          <img src={data.iconSrc} alt={data.label} />
        </ImageWrapper>
      )}
      <span>{data && data.label}</span>
    </OptionWrapper>
  );
};
