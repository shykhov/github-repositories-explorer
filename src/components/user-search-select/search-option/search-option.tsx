import React, { FC } from 'react';

import { OptionWrapper, ImageWrapper } from './search-option.styled';
import { SelectOption } from '../../../utils';

export interface Props {
  data: SelectOption;
  innerRef: React.Ref<HTMLDivElement>;
  innerProps: unknown;
  cx: string;
}

export const SearchOption: FC<Props> = props => {
  const { cx, innerProps, innerRef: ref, data, ...rest } = props;

  return (
    <OptionWrapper ref={ref} {...rest} {...innerProps}>
      {data && data.iconSrc && (
        <ImageWrapper>
          <img src={data.iconSrc} alt={data.label} />
        </ImageWrapper>
      )}
      <span>{data && data.label}</span>
    </OptionWrapper>
  );
};
