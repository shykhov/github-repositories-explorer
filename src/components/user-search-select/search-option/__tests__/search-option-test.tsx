import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';

import { SearchOption, Props } from '../search-option';

describe('<SearchOption />', () => {
  let renderedComponent: ShallowWrapper;
  let propsMock: Props;

  beforeEach(() => {
    propsMock = {
      data: { iconSrc: 'icon', value: 'test-value', label: 'test-label' },
      innerRef: React.createRef<HTMLDivElement>(),
      innerProps: {},
      cx: '',
    };

    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    renderedComponent = shallow(<SearchOption {...propsMock} />);
  });

  it('should render SearchOption', () => {
    expect(renderedComponent.dive()).toMatchSnapshot();
  });
});
