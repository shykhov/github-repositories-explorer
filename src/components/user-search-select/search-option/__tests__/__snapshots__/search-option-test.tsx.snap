// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchOption /> should render SearchOption 1`] = `
.c0 {
  cursor: pointer;
  padding: 7px 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:hover {
  background-color: #e6f2ff;
}

.c0 > span {
  margin-left: 10px;
  font-size: 16px;
}

<div
  className="c0"
>
  <styled.div>
    <img
      alt="test-label"
      src="icon"
    />
  </styled.div>
  <span>
    test-label
  </span>
</div>
`;
